tests/trainer/test_optimizers.py

```diff
@@ -335,25 +354,155 @@ def test_configure_optimizers_with_frequency(tmpdir):
     model = EvalModelTemplate()
     model.configure_optimizers = model.configure_optimizers__multiple_optimizers_frequency
 
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        max_epochs=1
-    )
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
     result = trainer.fit(model)
     assert result
 
 
+def test_step_scheduling_for_multiple_optimizers_with_frequency(tmpdir):
+    """
+    Test that step LR schedulers for multiple optimizers follow
+    the optimizer frequencies when corresponding frequency is set.
+    """
+
+    class DummyStepModel(BoringModel):
+        def training_step(self, batch, batch_idx, optimizer_idx):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            return {"loss": loss}
```

Comment:
```suggestion
            super().training_step(batch, batch_idx)
```

same for the tests below